### PR TITLE
spanner-jdbc: Add jdbc statement integration tests

### PR DESCRIPTION
**Pull request by olavloite**
Adds integration tests for JDBC statements and prepared statements.

### CODE REVIEW

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/resources/com/google/cloud/spanner/jdbc/it/Singers.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...oJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBrc96XvRx/YfuhN0f0nr7w+ur/1D34W3X/3zx+I7h885cXo/p3XDET3T/3UMdH9+9ZOje5fvOXy6P7dt384uj/tBwdG92fNeDu6v2veG9H9M762Iro/4bjHovsuAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBIAgFICAFBKAABKCQBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUEoAAEoJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACg1MCIhf8XfeAP/2s4uv/crr+L7k895J7o/oxDbo7uf3fm0uj+uaN+Et1fMfDD6P4FU7J//8b1b0f3D/3mQ9H90RecHt0f89G9o/vPf2lJdP/ji3dG9289OLvvAgAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBIAgFICAFBKAABKCQBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUEoAAEoJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASg186qzx0Qfe2bFfdP/B9306uv/5954T3f/iTx+O7l/yyKXR/YETn4ruH/1Pc6P7/3zyFdH9Px15RHR/8XuPje4PrZwd3T/t7KXR/f+dvCq6f9VbL0b33z3+19F9FwBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUEoAAEoJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBIAgFICAFBKAABKCQBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUGpw4iUfiz6wYtXY6P7s7zwf3d9xxZLo/titM6P7mz+R3Z8246Lo/jur1kf3D56+Mrp//u7d0f2Fd94f3b//6zui+xsn3xzdv3bH6dH9cXuOiO7f+p8jo/suAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBIAgFICAFBKAABKCQBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUEoAAEoJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACg1P8HAAD//+SLeeht2MttAAAAAElFTkSuQmCC'),DATE '1906-04-28');
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/resources/com/google/cloud/spanner/jdbc/it/Singers.txt]: As discussed, let's switch the BASE64 values to something generated by `RandomResultSetGenerator`.

Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/resources/com/google/cloud/spanner/jdbc/it/Singers.txt]: The BASE64 values have been replaced with generated random numbers.

**Comment by olavloite** [comment]
Test failure is caused by flake that is addressed by #6095.

**Comment by kolea2** [comment]
LGTM after rebase

**Comment by olavloite** [comment]
Rebased.

**Pull request merged**